### PR TITLE
Fixed #24701 -- Converted model manager names to unicode in migrations

### PR DESCRIPTION
**Pull request by MarkusH**
Ticket: https://code.djangoproject.com/ticket/24701

### CODE REVIEW

File: django/db/migrations/state.py [hunk 1]
```diff
@@ -448,16 +448,17 @@ def reconstruct_manager(mgr):
                 instance = manager_class(*args, **kwargs)
             # We rely on the ordering of the creation_counter of the original
             # instance
-            managers[mgr.name] = (mgr.creation_counter, instance)
+            name = force_text(mgr.name, strings_only=True)
```
Comment (timgraham) [inline django/db/migrations/state.py]: what's the purpose of strings_only here?

File: tests/migrations/test_state.py [hunk 2]
```diff
@@ -144,6 +145,7 @@ class Meta:
 
         # The default manager is used in migrations
         self.assertEqual([name for name, mgr in food_state.managers], ['food_mgr'])
+        self.assertEqual([type(name) for name, mgr in food_state.managers], [six.text_type])
```
Comment (timgraham) [inline tests/migrations/test_state.py]: I think I would use `assertTrue(all(isinstance(name, six.text_type) for ...)` but don't care too much either way.

File: django/db/migrations/state.py (hunk 1 above)
Comment (MarkusH) [inline django/db/migrations/state.py]: I went with the same pattern @claudep used in [#22564](https://code.djangoproject.com/ticket/22564).

File: tests/migrations/test_state.py (hunk 2 above)
Comment (MarkusH) [inline tests/migrations/test_state.py]: Yeah, that looks much better.

File: django/db/migrations/state.py (hunk 1 above)
Comment (claudep) [inline django/db/migrations/state.py]: AFAIR it is needed when the targeted value can be null, so as None is not transformed to the string "None". I don't know if it is the case here.

Comment (MarkusH) [inline django/db/migrations/state.py]: Yeah, makes sense. I just dropped the flag. There shouldn't be a way `manager.name` ever being `None` in this context.

**Comment by MarkusH** [comment]
Merged in faad6070ee2eeea270c76381f9ca5999bf1ac15f

**Pull request closed without merging**